From 6a047831103bc5404b779f668cfa984738a718fa Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Wed, 12 Mar 2014 09:50:08 +0000
Subject: [PATCH 32/33] coverity scan: fix leaking dir

Close dir before returning on error exit path, fixes leak detected
by coverity scan.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 src/thd_engine_default.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/thd_engine_default.cpp b/src/thd_engine_default.cpp
index f1158ef..46badd0 100644
--- a/src/thd_engine_default.cpp
+++ b/src/thd_engine_default.cpp
@@ -117,6 +117,7 @@ int cthd_engine_default::read_thermal_sensors() {
 									temp_input_str.str(), SENSOR_TYPE_RAW);
 							if (sensor->sensor_update() != THD_SUCCESS) {
 								delete sensor;
+								closedir(dir);
 								return THD_ERROR;
 							}
 							sensors.push_back(sensor);
-- 
1.9.0

